File: AOJ/1616.py
while 1:
    n, m = map(int, input().split())
    if n == 0:
        break

    a = list(map(int, input().split()))

    ans = 0
    for i in range(n):
        for j in range(i+1, n):
            if m < a[i] + a[j]:
                continue

            ans = max(ans, a[i] + a[j])

    if ans == 0:
        print("NONE")
    else:
        print(ans)

File: AOJ/1616.cpp
#include <bits/stdc++.h>
#define chmin(a, b) ((a)=min((a), (b)))
#define chmax(a, b) ((a)=max((a), (b)))
#define fs first
#define sc second
#define eb emplace_back
using namespace std;

typedef long long ll;
typedef pair<int, int> P;
typedef tuple<int, int, int> T;

const ll MOD=1e9+7;
const ll INF=1e18;

int dx[]={1, -1, 0, 0};
int dy[]={0, 0, 1, -1};

int main(){
    int n, m;
    while(cin>>n>>m, n){
        vector<int> a(n);
        for(int i=0; i<n; i++) cin>>a[i];

        int ans=-1;
        for(int i=0; i<n; i++){
            for(int j=i+1; j<n; j++){
                int sum=a[i]+a[j];
                if(sum<=m){
                    chmax(ans, sum);
                }
            }
        }

        if(ans==-1) cout << "NONE" << endl;
        else cout << ans << endl;
    }
}